spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerTemplate.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.secretmanager;
+
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.secretmanager.v1beta1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1beta1.AddSecretVersionRequest;
+import com.google.cloud.secretmanager.v1beta1.CreateSecretRequest;
+import com.google.cloud.secretmanager.v1beta1.ProjectName;
+import com.google.cloud.secretmanager.v1beta1.Replication;
+import com.google.cloud.secretmanager.v1beta1.Secret;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient.ListSecretsPagedResponse;
+import com.google.cloud.secretmanager.v1beta1.SecretName;
+import com.google.cloud.secretmanager.v1beta1.SecretPayload;
+import com.google.cloud.secretmanager.v1beta1.SecretVersionName;
+import com.google.protobuf.ByteString;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+
+/**
+ * Offers convenience methods for performing common operations on Secret Manager including
+ * creating and reading secrets.
+ *
+ * @author Daniel Zou
+ * @since 1.3
+ */
+public class SecretManagerTemplate implements SecretManagerOperations {
+
+	private final SecretManagerServiceClient secretManagerServiceClient;
+
+	private final GcpProjectIdProvider projectIdProvider;
+
+	public SecretManagerTemplate(
+			SecretManagerServiceClient secretManagerServiceClient,
+			GcpProjectIdProvider projectIdProvider) {
+		this.secretManagerServiceClient = secretManagerServiceClient;
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	@Override
+	public void createSecret(String secretId, String payload) {
```

Comment:
since we know whether a secret is new or existing, would it make sense to return a boolean flag, an enum or a version number indicating the result?

Reply:
I prefer `void` just to keep it simple for now; but definitely open to implementing this if users request for it in the future.